### PR TITLE
feat: Add Dashboard pipeline

### LINKED ISSUE #334: [Dashboard] Add pipeline for running test cases
1. Before applying, Read our [Contributor Guidelines](https://github.com/CarmineOptions/derisk-research/blob/master/CONTRIBUTING.md) and [README](https://github.com/CarmineOptions/derisk-research/blob/master/README.md). 
2. Create github pipeline for running test cases in `dashboard_app`
3. Please create one simple test file to test it using pytest
4. Feel free to check how it was done for another app, [`data_handler`](https://github.com/CarmineOptions/derisk-research/blob/master/.github/workflows/ci.yml)

### PR DESCRIPTION
**Pull request by No-bodyq**
resolves #334

### CODE REVIEW

**Comment by No-bodyq** [comment]
Ready for review @djeck1432

**Review by djeck1432 — CHANGES_REQUESTED** [review]
```
_____________________ ERROR collecting tests/main_test.py ______________________
ImportError while importing test module '/home/runner/work/derisk-research/derisk-research/apps/dashboard_app/tests/main_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/main_test.py:3: in <module>
    from dashboard_app.charts import Dashboard
charts/__init__.py:1: in <module>
    from dashboard_app.charts.main import Dashboard
charts/main.py:3: in <module>
    from dashboard_app.charts.main_chart_figure import get_main_chart_figure
charts/main_chart_figure.py:15: in <module>
    from shared.state import State
../shared/state.py:8: in <module>
    from shared.error_handler import BOT, MessageTemplates, TokenSettingsNotFound
../shared/error_handler/__init__.py:2: in <module>
    from .notifications import BOT
../shared/error_handler/notifications.py:5: in <module>
    from aiogram import Bot, Dispatcher
E   ModuleNotFoundError: No module named 'aiogram'
=============================== warnings summary ===============================
../../../../../.cache/pypoetry/virtualenvs/derisk-research-o1wA9558-py3.11/lib/python3.11/site-packages/starknet_py/abi/v1/schemas.py:[28](https://github.com/No-bodyq/derisk-research/actions/runs/12059071047/job/33627010142#step:7:29)
  /home/runner/.cache/pypoetry/virtualenvs/derisk-research-o1wA9558-py3.11/lib/python3.11/site-packages/starknet_py/abi/v1/schemas.py:28: RemovedInMarshmallow4Warning: The 'default' argument to fields is deprecated. Use 'dump_default' instead.
    state_mutability = fields.String(data_key="state_mutability", default=None)

../../../../../.cache/pypoetry/virtualenvs/derisk-research-o1wA9558-py3.11/lib/python3.11/site-packages/starknet_py/abi/v2/schemas.py:[34](https://github.com/No-bodyq/derisk-research/actions/runs/12059071047/job/33627010142#step:7:35)
  /home/runner/.cache/pypoetry/virtualenvs/derisk-research-o1wA9558-py3.11/lib/python3.11/site-packages/starknet_py/abi/v2/schemas.py:34: RemovedInMarshmallow4Warning: The 'default' argument to fields is deprecated. Use 'dump_default' instead.
    state_mutability = fields.String(data_key="state_mutability", default=None)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/main_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
````

LOoks to me,that you frogot to install `shared` package

**Comment by No-bodyq** [comment]
Hello @djeck1432, I added a step to install dependencies for the shared package but i am still getting the same error. Could i get some assistance on this??

**Comment by djeck1432** [comment]
@No-bodyq Did you look at `data_handler` pipeline, where it was implemented already?

**Review by djeck1432 — CHANGES_REQUESTED** [review]
1. sync with master branch
2. Add to `dashboard_app` new package using `poetry add <missing> package
3. feel free to add also `poetry.lock` file here

**Comment by No-bodyq** [comment]
@djeck1432 I am getting this error 
{9A6DF2B1-A86D-4F4A-B531-942F54F1A7F2}
When i run the poetry add package command

**Comment by djeck1432** [comment]
@No-bodyq still see this error:
```
ImportError while importing test module '/home/runner/work/derisk-research/derisk-research/apps/dashboard_app/tests/test_state.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.11.10/x[64](https://github.com/No-bodyq/derisk-research/actions/runs/12070541863/job/33660261612#step:8:65)/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_state.py:2: in <module>
    from shared.state import State
../shared/state.py:8: in <module>
    from shared.error_handler import BOT, MessageTemplates, TokenSettingsNotFound
../shared/error_handler/__init__.py:2: in <module>
    from .notifications import BOT
../shared/error_handler/notifications.py:5: in <module>
    from aiogram import Bot, Dispatcher
E   ModuleNotFoundError: No module named 'aiogram'
```

regarding what you shared with me, it's error which you should fix, maybe something bad copy/pasted and now the line is broken. You can use google to look up the solution

File: apps/dashboard_app/tests/main_test.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import pytest
```
Comment (djeck1432) [inline apps/dashboard_app/tests/main_test.py:1]: remove this file, there is already some test cases

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**